scripts/exregional_run_fcst.sh

```diff
@@ -533,6 +533,27 @@ export KMP_AFFINITY=scatter
 export OMP_NUM_THREADS=${OMP_NUM_THREADS:-1} #Needs to be 1 for dynamic build of CCPP with GFDL fast physics, was 2 before.
 export OMP_STACKSIZE=1024m
 
+#
+#-----------------------------------------------------------------------
+#
+# If INPUT/phy_data.nc exists, convert it from NetCDF4 to NetCDF3
+# (happens for cycled runs, not cold-started)
+#
+#-----------------------------------------------------------------------
+#
+cd INPUT
+if [[ -f phy_data.nc ]] ; then
+  rm -f phy_data.nc3 phy_data.nc4
+  cp -fp phy_data.nc phy_data.nc4
+  if ( ! time ( module purge ; module load intel szip hdf5 netcdf nco ; module list ; set -x ; ncks -3 --64 phy_data.nc4 phy_data.nc3) ) ; then
+    mv -f phy_data.nc4 phy_data.nc
+    rm -f phy_data.nc3
+    echo "NetCDF 3=>4 conversion failed. :-( Continuing with NetCDF 4 data."
+  else
+    mv -f phy_data.nc3 phy_data.nc
+  fi
+fi
+cd ..
 #
```

Comment:
@SamuelTrahanNOAA, can you explain how the ncks command is run for the "else" portion of the inner "if" statement?  I can't see how that happens.  Thanks.

Reply:
The `if` line runs the command. After the `time ( module purge ; module load intel szip hdf5 netcdf nco ; module list ; set -x ; ncks -3 --64 phy_data.nc4 phy_data.nc3)` runs, the conversion, or failure to convert, has happened. Then the `if ( ! ... )` checks if that command failed by looking at the return status. When it fails (non-zero status becomes zero from !), the first block runs, reverting to the NetCDF4 file. When it succeeds (zero status becomes one from !), it falls into the "else" block, where the NetCDF3 file becomes the phy_data.nc.

Reply:
Thanks, Sam!